src/Parser.php

```diff
@@ -54,9 +44,9 @@ class Parser
      */
     public function isScheme(string $scheme): bool
     {
-        return '' === $scheme
-            || (strlen($scheme) === strspn($scheme, self::SCHEME_VALID_CHARS)
-                && false !== strpos(self::SCHEME_VALID_STARTING_CHARS, $scheme[0]));
+        static $pattern = '/^[a-z][a-z\+\.\-]*$/i';
```

Comment:
do static vars offer a perf benefit over class constants?

Reply:
This has just been done, so they are not public API.

Reply:
ok so actually this is a BC break.

Reply:
With constants it's a BC break, too. But we can re-add the old constants and mark them as deprecated.

Reply:
the constants can be re-introduced and left as is to avoid any BC break if they are not used anymore. if it was a PHP7.1+ library they would have been marked as private. I did not documented for that reason on the documentation website.

Reply:
@nyamsprod Better mark such places with `/** @internal */` next time, so people can be sure it's internal and not public API, largely avoids this issue.

Making the class `final` and all methods `private` also help reducing such things being BC breaks.

Reply:
@kelunik agreed like I said this whole class should have been final with private methods only. It will be like that for the next major release.